### PR TITLE
adding s3 dataset support

### PR DESCRIPTION
**Pull request by sudeshsh**

### CODE REVIEW

File: kubeflow/tf-job-e2e/dataset.libsonnet [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      metadata: {
+        labels: {
+          app: "dataset",
+        },
+      },
+      spec: {
+        containers: [
+          {
+            name: "dataset",
+            image: "vladgh/s3sync",
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/dataset.libsonnet]: this should be a variable we dont want to use embedded images in a prototype.

File: kubeflow/tf-job-e2e/dataset.libsonnet [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                name: "AWS_SECRET_ACCESS_KEY",
+                valueFrom: {
+                  secretKeyRef: {
+                    name: "aws-secret",
+                    key: "awsSecretAccessKey",
+                  },
+                },
+              },
+              {
+                name: "AWS_DEFAULT_REGION",
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/dataset.libsonnet]: do we need this in a job?

File: kubeflow/tf-job-e2e/prototypes/dataset-download-s3.jsonnet [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    std.split(argsParam, ",");
+
+local aws_key = import "param://aws_key";
+local aws_secret = import "param://aws_secret";
+local storage_class = import "param://storage_class";
+local s3Path = import "param://s3Path";
+local syncdir = import "param://syncdir";
+
+std.prune(k.core.v1.list.new([
+  dataset.parts.secret(namespace,aws_key,aws_secret),
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/prototypes/dataset-download-s3.jsonnet]: add spaces between args like the one below

File: kubeflow/tf-job-e2e/prototypes/dataset-download-s3.jsonnet [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  dataset.parts.pvc(namespace, storage_class),
+  dataset.parts.job(namespace,s3Path,syncdir),
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/prototypes/dataset-download-s3.jsonnet]: same here

File: kubeflow/tf-job-e2e/prototypes/tf-job-e2e-s3.jsonnet [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+local configMapName = import "param://config_map_name";
+local image = import "param://image";
+local imageGpu = import "param://image_gpu";
+local numMasters = import "param://num_masters";
+local numPs = import "param://num_ps";
+local numWorkers = import "param://num_workers";
+local numGpus = import "param://num_gpus";
+local dataDir = import "param://data_dir";
+local pvc = {
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/prototypes/tf-job-e2e-s3.jsonnet]: definition of a pvc shape should be in the libsonnet and in the prototype you call the function with the params

File: kubeflow/tf-job-e2e/prototypes/tf-job-e2e-s3.jsonnet [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            claimName: "dataset"
+          },
+        };
+local volumeMounts = [ 
+  mount.new("train-dir-vol","/tmp"),
+   mount.new("dataset-storage",dataDir,true)
+];
+local volumes = [
+  volume.fromConfigMap("train-dir-vol", configMapName, {}),
+  (pvc)
```
Comment (raddaoui) [inline kubeflow/tf-job-e2e/prototypes/tf-job-e2e-s3.jsonnet]: why did you put pvc between ()?

File: kubeflow/tf-job-e2e/dataset.libsonnet (hunk 1 above)
Comment (sudeshsh) [reply kubeflow/tf-job-e2e/dataset.libsonnet]: we are gonna use the same image for data download, i don't think we need to put a variable here.

File: kubeflow/tf-job-e2e/dataset.libsonnet (hunk 2 above)
Comment (sudeshsh) [reply kubeflow/tf-job-e2e/dataset.libsonnet]: removing

**Pull request merged**